BaseTools/Plugin/UefiVarPatcher/UefiVarPatcher.py

```diff
@@ -0,0 +1,133 @@
+# @file UefiVarPatcher.py
+# Plugin used to updated UEFI variables in a ROM image
+#
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+import logging
+import os
+
+import edk2toollib.uefi.edk2.variablestore_manulipulations as VarStore
+import edk2toollib.uefi.edk2.variable_format as VF
+
+from edk2toolext.environment.plugintypes.uefi_build_plugin import IUefiBuildPlugin
+from edk2toolext.environment.uefi_build import UefiBuilder
+from edk2toolext.uefi.patch_var_store import (
+    load_variable_xml,
+    patch_variables,
+    create_variables,
+)
+
+PLUGIN_NAME = "UefiVarPatcher"
+CONFIG_FILE_NAME = "BuiltInVars.xml"
+
+
+class UefiVarPatcher(IUefiBuildPlugin):
+    def do_post_build(self, builder: UefiBuilder) -> int:
+        """UEFI variable patching post-build functionality.
+
+        Args:
+            builder (UefiBuilder): A UEFI builder object for this build.
+
+        Returns:
+            int: Returrn code. 0 for success, -1 for failure.
+        """
+        logging.info(f"PLUGIN {PLUGIN_NAME}: Patching initial NvStore contents")
+
+        # Not an error; no MSFT_PLATFORM_PACKAGE defined, so don't try to apply built-in vars.

```

Comment:
even though this will be a breaking change, we need to change the name of this env to not include MSFT.

Reply:
Done